### PR TITLE
SCT-2935 multiple search terms

### PR DESCRIPTION
**Pull request by eapearson**
# Fix SCT-2935 concerning multiple search terms not narrowing the search

The effective code change for this fix is a single line, which adds the `AND` operator to the search params for the search public data source.

The bulk of the rest of the changes relate to test configuration, test code, and test support, to better provide for testing of this and related tests.

The issue is related to a public ticket, PUBLIC-1517, and related tickets. This is one of the issues brought up in that ticket. Other related issues are under different tracking tickets  and will be separate PRs.

# Jira Ticket / Issue #1684 

- SCT-2935 for this fix
- PUBLIC-1517 which motivated and inspired this fix.

- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
- [x] Tests pass locally and in GitHub Actions
- [x] Changes available by spinning up a local narrative and navigating to the refseq public data tab and conducting a search with a binomial scientific name

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] (JavaScript) I have run Prettier and ESLint on changed code manually or with a git precommit hook
- [-] (Python) I have run Black and Flake8 on changed Python code manually or with a git precommit hook

# Updating Version and Release Notes (if applicable)

- [-] [Version has been bumped](https://semver.org/) for each release
- [x] [Release notes](/RELEASE_NOTES.md) have been updated for each release (and during the merge of feature branches)

### CODE REVIEW

File: kbase-extension/static/kbase/js/widgets/narrative_core/kbaseNarrativeSidePublicTab.js [hunk 1]
```diff
@@ -498,39 +481,48 @@ define ([
...
-                var maxPage = Math.ceil(this.currentFilteredResults / this.itemsPerPage);
+                const maxPage = Math.ceil(this.currentFilteredResults / this.itemsPerPage);
                 if (this.currentPage >= maxPage) {                    
                     return;
                 }
             }
 
+            // We use a flag here as a lock on rendering more, so that a renderMore
+            // request will complete before advancing the page and firing off another
+            // data fetch and render request.
```
Comment (briehl) [inline kbase-extension/static/kbase/js/widgets/narrative_core/kbaseNarrativeSidePublicTab.js:492]: :+1:

**Review by briehl — CHANGES_REQUESTED** [review]
Looks like the main changes were in the data source and sidePublicTab to add the AND operator to multiple search terms, as well as a bunch of little modernization niceties. Looks good to me, but I have a couple minor details before merging.

File: test/integration/specs/kbaseNarrativeSidePublicTab_test.js [hunk 2]
```diff
@@ -316,38 +463,96 @@ async function waitForRows(panel, count){
...
+async function getRow(publicPanel, testCase) {
+    const rows = await waitForRows(publicPanel, testCase.row);
+    expect(rows.length).toBeGreaterThanOrEqual(testCase.row);
+    const row = rows[testCase.row - 1];
+    expect(row).toBeDefined();
+    return row;
+}
+
+async function scrollToRow(publicPanel, testCase) {
+     const rows = await waitForRows(publicPanel, testCase.row);
```
[the remark this answers is not in the document]
Comment (eapearson) [reply test/integration/specs/kbaseNarrativeSidePublicTab_test.js]: done.

File: docs/integration-testing.md [hunk 3]
```diff
@@ -484,12 +484,12 @@ BROWSER=safari PRESET=bs-mac-chrome make test-integration
 ##### To test a different OS Version
 
 ```bash
-OS_VERSION="Mojave" PRESET=bs-mac-firefox make test-integration
+OS_VERSION="Mojave" PRESET=bs-mac-FireFox make test-integration
```
[the remark this answers is not in the document]
Comment (eapearson) [reply docs/integration-testing.md]: done

File: docs/integration-testing.md [hunk 4]
```diff
@@ -451,16 +451,16 @@ export KBASE_TEST_TOKEN={MYTOKEN}
 export SERVICE_USER={MYSERVICEUSER}
 export SERVICE_KEY={MYSERVICEKEY}
 PRESET=bs-mac-chrome make test-integration
-PRESET=bs-mac-firefox make test-integration
+PRESET=bs-mac-FireFox make test-integration
```
[the remark this answers is not in the document]
Comment (eapearson) [reply docs/integration-testing.md]: done

**Comment by eapearson** [comment]
@briehl I took care of the issues pointed out in comments, and also resolved the merge conflicts that crept in.

**Pull request merged**